app.go

```diff
@@ -85,3 +88,11 @@ func Automatic(opts Options) *App {
 
 	return a
 }
+
+func (a *App) addDefaultRouteNames() {
```

Comment:
@markbates I'll try to explain myself a bit better, sorry for the confusion, but first let me mention some constraints i've understood we have on this feature:

1. We want to use gorilla's mux named features
2. We don't want to add more code to the actions.go file

To start, right now we have the following code to add a route into our application:

```go
app.GET("/", SomeHandler)
```

What i was mentioning before is that it seems that we cannot add the default route name on the mux Route thats inside the RouteInfo (if none passed) by the time GET/POST(...) method is called because by doing so we would be disallowing the RouteName method to add the route and invalidating the entire route (because of [this](https://github.com/gorilla/mux/blob/34dda716af12ba79ed40ce1f66b102ff75dc3411/route.go#L115)).

I agree that this happening on every request is not good but i still looking for a better place to do it without either adding more code into the generated actions.go or storing the route name outside the *mux.Route.

i hope this makes more sense!